src/dune_pkg/solver_env.ml

```diff
@@ -54,6 +54,18 @@ module Variable = struct
         of_ordered_set ordered_set
       ;;
 
+      let encode t =
+        Dune_sexp.List
+          (to_list t
+           |> List.map ~f:(fun flag ->
+             (* Note that flags are quoted to make the output a valid Ordered
+                Set Language expression so that [encode] and [decode] can round
+                trip. Technically only the first flag needs to be quoted to
+                form a valid OSL expression but all are quoted for
+                consistency.*)
+             Dune_sexp.Quoted_string (to_string flag)))
```

Comment:
I wonder whether it would make sense to implement `encode` on `Ordered_set_lang` to be able to write the (unevaluated) expression back to the lockfile.

On one hand it is nice to see what exactly the OSL has evaluated to, but on the other hand there is a not so nice part where we expose `:standard` and the expression in the lockfile doesn't match the expression from the dune-workspace.

This is very much a thinking out loud thing, let me know if you have any arguments for or against either of these behaviours.

Reply:
`:standard` is specific to the version of the dune language (used in the workspace file). While the lock directory should be stable wrt to its own version. For that reason, we should evaluate it.

Reply:
I think it's ok that the expression in the lockfile doesn't match the expression in the workspace file. This is really just so human readers can get a sense for the conditions under which the lockdir was generated.